usr/share/rear/build/GNU/Linux/400_copy_modules.sh

```diff
@@ -1,28 +1,129 @@
+        for loaded_module_file in $loaded_modules_files ; do
+            if ! cp $verbose -t $ROOTFS_DIR -L --preserve=all --parents $loaded_module_file 1>&2 ; then
+                Error "Failed to copy $loaded_module_file"
+            fi
+        done
+        # After successful copying do the the code after the artificial 'for' clause:
+        continue
+    fi
+
+    # Finally the default case:
+    LogPrint "Copying kernel modules"
+    for module in "${MODULES[@]}" ; do
+        # Strip trailing ".o" if there:
+        module=${module#.o}
+        # Strip trailing ".ko" if there:
+        module=${module#.ko}
+        # Check if the module is not in the exclude list:
+        for exclude_module in "${EXCLUDE_MODULES[@]}" ; do
+            # Continue with the next module if the current one is in EXCLUDE_MODULES:
+            test "$module" = "$exclude_module" && continue 2
+        done
+        # Continue with the next module if the current one does not exist:
+        modinfo $module &>/dev/null || continue
+        # Use /etc/modprobe.conf if exists:
+        with_modprobe_conf=""
+        test -e /etc/modprobe.conf && with_modprobe_conf="-C /etc/modprobe.conf"
+        # Resolve module dependencies:
+        # I.e. get the module file plus the module files of other needed modules.
+        # This is currently only a "best effort" attempt because
+        # in general 'modprobe --show-depends' insufficient to get all needed modules
+        # see https://github.com/rear/rear/issues/1355
+        # The --ignore-install is helpful because that converts currently unsupported '^install' output lines
+        # into supported '^insmod' output lines for the particular module but that is also insufficient
+        # see also https://github.com/rear/rear/issues/1355
+        module_files=$( /sbin/modprobe $with_modprobe_conf --ignore-install --set-version $KERNEL_VERSION --show-depends $module 2>/dev/null | awk '/^insmod / { print $2 }' )
+        if ! test "$module_files" ; then
+            # Fallback is the plain module file without other needed modules (cf. the MODULES=( 'loaded' ) case above):
+            # Can it really happen that a module exists (which is tested above) but 'modinfo -F filename' cannot show its filename?
+            # To be on the safe side there is a test even for for such a possibly weird error here:
+            module_files="$( modinfo -F filename $module || Error "$module exists but no module file?" )"
+        fi
+        for module_file in $module_files ; do
+            if ! cp $verbose -t $ROOTFS_DIR -L --preserve=all --parents $module_file 1>&2 ; then
+                Error "Failed to copy $module_file"
+            fi
+        done
+    done
 
-# copy modules & depmod
-LogPrint "Copying kernel modules"
-ModulesCopyTo "$ROOTFS_DIR" "${MODFILES[@]}" >&8
```

Comment:
Is ModulesCopyTo now obsolete? Then please remove it. However, I would suggest to extract such things into functions if it can be tested independently of the other script or used elsewhere.